### PR TITLE
Missing profile picture creates request for 404

### LINKED ISSUE #59: Missing profile picture creates request for `404`
### Describe the bug

When a user does not have a picture set (`undefined` or `null`), there is a request to `undefined` which goes `404`

Here:

https://github.com/tupuio/tupuapp/blob/a7f118ef4e04efe7cca81634ab24fb46c0e3bc2b/components/RequestsList/RequestCard.js#L43

And here:

https://github.com/tupuio/tupuapp/blob/a7f118ef4e04efe7cca81634ab24fb46c0e3bc2b/components/ApplicationsList/ApplicationCard.js#L41

### Suggested solution

https://github.com/tupuio/tupuapp/blob/a7f118ef4e04efe7cca81634ab24fb46c0e3bc2b/components/MentorsList/MentorCard.js#L36-L46

### To Reproduce

1. Open the Dev Tools, network analyzer
1. Go to requests list with a mentee without picture
1. Notice the `404` on an `undefined?tr=w-100,h-100,fo-auto`

### Expected behavior

A placeholder image is loaded (e.g. https://robohash.org/your_text_here.png?size=100x100 ), or no request is made.

### Actual behavior

An http request is made, it fails with 404. A placeholder is shown with the initial letter from the name of the mentor/mentee

### PR DESCRIPTION
**Pull request by SamuelRott**
### What Github issue does this PR relate to?

Close #59

### Changes proposed in this pull request:

Use ternary operator to show Avatar component with or without `src` attribute

### Quality Assurance(QA) Checklist

- [x] The app builds and runs properly (if this means locally, then yes)

### CODE REVIEW

File: components/ApplicationsList/ApplicationCard.js [hunk 1]
```diff
@@ -36,12 +36,20 @@ const ApplicationCard = ({ application }) => {
...
+                  name={application.mentee.name}
+                  alignSelf={"center"}
+              />
+          ) : (
+              <Avatar
+                  size="xl"
+                  name={application.mentee.name}
+                  alignSelf={"center"}
+              />
+          )}
```
Comment (lorepirri) [inline components/ApplicationsList/ApplicationCard.js:52]: Congratulations on your first PR!

The two `true`/`false` branches have a lot in common, maybe we could simplify to just have an empty `src` if not `application.mentor.picture`?

The network request should not be triggered in that case.

Removing the `src` if not needed also could be a viable solution.

File: components/RequestsList/RequestCard.js [hunk 2]
```diff
@@ -39,12 +38,20 @@ const RequestCard = ({ request, handleContact, handleAccept, handleReject }) =>
...
-          <Avatar
-            size="xl"
-            src={request.mentee.picture + "?tr=w-100,h-100,fo-auto"}
-            name={request.mentee.name}
-            alignSelf={"center"}
-          />
+          {request.mentor.picture ? (
+              <Avatar
+                  size="xl"
+                  src={request.mentee.picture + "?tr=w-100,h-100,fo-auto"}
```
Comment (lorepirri) [inline components/RequestsList/RequestCard.js:44]: same here as above

**Review by lorepirri — CHANGES_REQUESTED** [review]

File: components/ApplicationsList/ApplicationCard.js (hunk 1 above)
Comment (lorepirri) [reply components/ApplicationsList/ApplicationCard.js:52]: also, is it `application.mentor` or `application.mentee` here?

File: components/RequestsList/RequestCard.js (hunk 2 above)
Comment (lorepirri) [reply components/RequestsList/RequestCard.js:44]: `application.mentor` / `application.mentee`

File: components/ApplicationsList/ApplicationCard.js (hunk 1 above)
Comment (SamuelRott) [reply components/ApplicationsList/ApplicationCard.js:52]: Ouch, being caught copy/pasting on the first PR...

> The two true/false branches have a lot in common, maybe we could simplify to just have an empty src if not application.mentor.picture?

I had a look at the Avatar doc from Chakra and I did not see the empty `src` has a solution, but maybe it is an obvious solution, I will try and see how it rect with empty `src`

Comment (lorepirri) [reply components/ApplicationsList/ApplicationCard.js:52]: > I had a look at the Avatar doc from Chakra and I did not see the empty `src` has a solution, but maybe it is an obvious solution, I will try and see how it rect with empty `src`

You can play around in the [doc page for the Avatar](https://chakra-ui.com/docs/components/avatar) which is editable, and you could see using the DevTools that there is no network request when the `src` is empty. To be tested in this situation, maybe the editable box behaves differently.

Comment (lorepirri) [reply components/ApplicationsList/ApplicationCard.js:52]: > Ouch, being caught copy/pasting on the first PR...

No worries ^^ request review again when you're ready ;)

**Comment by lorepirri** [comment]
taking over this

**Review by lorepirri — COMMENTED** [review]
@SamuelRott I took this over, I'm gonna merge soon, tagging you in case you are still interested in the solution. It seems to work.

avatar-src-empty

**Review by lorepirri — APPROVED** [review]

**Pull request merged**